### PR TITLE
To Do List Structure

### PR DESCRIPTION
**Pull request by anyars-encarta**
In this project,

- [ ] I setup a To Do List project with Webpack.
- [ ] I created an **index.html** file and wrote my HTML markup there. 
- [ ] I created an empty To Do List **ul** placeholder . 
- [ ] The **index.html** file is a template created with HTML Webpack Plugin.
- [ ] I created an **index.js** file and set an array of to do tasks. Each task object contains three keys: **ID**, **Description**, and **Boolean** value.
- [ ] I wrote a function to iterate over the tasks array and populate an HTML list item element for each task.
- [ ] On page load, the tasks are dynamically created. 
- [ ] The list appear in order of the index values for each task.
- [ ] Clicking on each Task toggles between the move icons (3 vertical dots on the right) and the delete icon (a trash can).
- [ ] I created a **style.css** and set rules for the To Do List. 
- [ ] The CSS is loaded by Webpack Style/CSS Loader.
- [ ] All the source files (**index.html, index.js and style.css**) are located in the **/src directory** and the distribution files are generated by webpack and served by **webpack dev server** from the **/dist** folder.
- [ ] The live server runs on port **3000**.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* eslint-disable object-shorthand */
+import './styles/style.css';
+import taskLoad from './modules/addTasks.js';
+import myDateTime from './modules/date-time.js';
+
+// Fetched from module
+document.addEventListener('DOMContentLoaded', () => {
+  taskLoad();
```
Comment (shahadat3669) [inline src/index.js:8]: 👍🏆 Your project looks good and functions well! However, I noticed that you completed the entire project in one milestone. While it's impressive, it's generally not the best approach. So please consider working on the specific milestone which will allow you to work on specific tasks,  request review and make changes upon review, and update them over time. 🔄📅

For this specific mission, a few adjustments are still needed:

- [ ] As per the project requirements, in the  index.js file set an array of some simple to-do tasks (array of objects). Each task object should contain three keys: description [string], completed [bool], and index [number]. 📝🗒️

- [ ] In the  index.js file, write a function to iterate over the tasks array and dynamically populate an HTML list item element for each task. 🔄📝📜

- [ ] Upon page load, render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task. 🔄🖥️

By making these adjustments and following the project requirements, your project will be even more polished and organized. Keep up the great work, and I'm excited to see how your project evolves through milestones! 🚀🎉

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @anyars-encarta , 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Descriptive PR Title
✅ Descriptive PR Description


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable object-shorthand */
```
Comment (V-Blaze) [inline src/index.js:1]: - [ ] You have done great implementing the suggestions from the previous CR and also meeting the project requirements :+1: :+1: , But there is a change you need to make before moving to the next milestone. :+1: 
- [ ] As required by Microverse, you are not meant to use any linters diable rule in your codebase. The linters are there to make sure your code meets the best standard practices. I'll suggest you remove this linter diable rule and fix the issue. :+1: :+1: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @anyars-encarta , 

# Almost there 🟡 

1094a3e620ebea4cd6cf2d0aceaa8f52


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing some changes from the previous reviewers, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
-  Webpack setup correctly✔
- App works as expected💯 🥇 ✔

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**